File: tdd-katas/gilded-rose-kata/gilded-rose-javascript/gilded_rose_test.js
require('approvals')
  .mocha();
var { Shop, Item } = require('./gilded_rose.js');

function convert_items_to_string(items) {
    data = ""
    for (i = 0; i < items.length; i++) 
      data += items[i].toString() + "\n"
    return data
}

const items = [
  new Item("+5 Dexterity Vest", 10, 20),
  new Item("Aged Brie", 2, 0),
  new Item("Elixir of the Mongoose", 5, 7),
  new Item("Sulfuras, Hand of Ragnaros", 0, 80),
  new Item("Sulfuras, Hand of Ragnaros", 1, 80),
  new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
  new Item("Backstage passes to a TAFKAL80ETC concert", 10, 51),
  new Item("Backstage passes to a TAFKAL80ETC concert", 5, 49),
  new Item("Conjured Mana Cake", 3, 6)
]

describe('Gilded Rose', function() {
  it('should output equal to snapshot', function() {
    const gildedRose = new Shop(items)
    
    gildedRose.updateQuality()
    gildedRose.updateQuality()
    const updated_items = gildedRose.updateQuality()    
    this.verify(convert_items_to_string(updated_items))
  })
})

File: kata-solutions/gilded-rose-kata/gilded-rose-python/1-golden-master/test/gilded_rose_test.py
import unittest
from approvaltests.combination_approvals import verify_all_combinations
from gilded_rose import GildedRose, Item


class GildedRoseTest(unittest.TestCase):
    def test_update_combinatoric(self):
        names = [
            "foo", "Aged Brie", "Sulfuras, Hand of Ragnaros",
            "Backstage passes to a TAFKAL80ETC concert"
        ]
        sellIns = [-1, 2, 6, 0, 11, 7]
        qualities = [0, 48, 49, 50, 47]
        verify_all_combinations(self.do_update_quality,
                                [names, sellIns, qualities])

    def do_update_quality(self, name: str, sellIn: int, quality: int) -> str:
        items = [Item(name, sellIn, quality)]
        app = GildedRose(items)
        app.update_quality()
        return app.items[0]


if __name__ == "__main__":
    unittest.main()
